### PR TITLE
:rocket: v9

### LINKED ISSUE #1938: [TODO] Prepare v9
Tracking list of changes for `v9`:

- [x] Drop support for `yarn`
- [x] Drop support for `node v7.9.0`
- [x] Drop support from `vscode` using `node v7.9.0`
- [x] Update dependencies
- [x] Use true `IoC` via `Inversify`
- [x] Do not show `Restart` messages for `vscode` versions that support icon theme reloading
- [x] Improve language resource manager to include a language as soon as it lands in our repo
- [x] Bundle the `nls` resources to a single file, wherever applicable
- [x] Mute console messages when running in `debug` mode
- [x] Show Error message if `vscode` version is not supported
- [x] Make all I/O operations asynchronous
- [x] Bundle application via `Webpack`
- [x] Add test case to prevent duplicate icon declarations without different `disabled` attribute
- [x] Implement application integrity check via `nsri`
- [x] Switch to new `CodeClimate` test reporter (`linux` and `osx` only)
- [x] Drop usage of `Codecov` test reporter
- [x] Improved project detection
- [x] Switch to new `vscode` typings (`@types/vscode`)
 
The above list is subjected to changes as we get along.

### PR DESCRIPTION
**Pull request by JimiC**
_**Closes #1938**_

**Changes proposed:**
- [x] Prepare

This PR contains the following changes:

- Drop support for `yarn`
- Drop support for `node v7.9.0`
- Drop support from `vscode` using `node v7.9.0`
- Update dependencies
- Use true `IoC` via `Inversify`
- Do not show `Restart` messages for `vscode` versions that support icon theme reloading
- Improve language resource manager to include a language as soon as it lands in our repo
- Bundle the `nls` resources to a single file, wherever applicable
- Mute console messages when running in `debug` mode
- Show Error message if `vscode` version is not supported
- Make all I/O operations asynchronous
- Bundle application via `Webpack`
- Add test case to prevent duplicate icon declarations without different `disabled` attribute
- Implement application integrity check via `nsri`
- Switch to new `CodeClimate` test reporter (`linux` and `osx` only)
- Drop usage of `Codecov` test reporter
- Improved project detection
- Switch to new `vscode` typings (`@types/vscode`)
 
Enjoy!

### CODE REVIEW

**Comment by JimiC** [comment]
Just want to mention something regarding the implementation of the integrity check. In case of a corrupted distro a message is displayed to the user warning him/her about it. This message will be presented every time the extension get's activated and CAN NOT be suppressed. This is intentional. The extension activation though will not be blocked. The extension will still be usable but no support will be provided for those cases.

File: src/services/compositionRootService.ts [hunk 1]
```diff
@@ -1,113 +1,144 @@
...
+      ],
+      [IntegrityManager, []],
+    ];
+  }
+
+  private initDecorations(): void {
+    this.injectableClasses.forEach(injectableClass => {
+      // declare classes as injectables
+      const klass: Class = injectableClass[0];
+      decorate(injectable(), klass);
```
Comment (KingDarBoja) [inline src/services/compositionRootService.ts]: This line is giving a test error:
`Cannot apply @Injectable decorator multiple times.`

Please refer to [this issue](https://github.com/inversify/InversifyJS/issues/563) ànd the author for a possible workaround.

**Comment by JimiC** [comment]
1. Regarding `CodeClimate` the new reporter does not support `WinOS`.  On the other hand, we only need to upload one report from the entire tets suite. I could have made it so that only one build reports but it would get complicated. Since the new reporter just produces a warning when another report had been already uploaded I kept it simple in the `travis` config file.

2. Turns out that switching from `Angular` to `NestJS` projects requires that the icons manifest needs to be reset. If you switch from `Angular` to `Non-Angular/Non-NestJS` to `NestJS` it works. Not sure how to tackle this, will need some time to ponder on it.

File: src/services/compositionRootService.ts (hunk 1 above)
Comment (JimiC) [reply src/services/compositionRootService.ts]: Is this happening when you use the `-w` switch?

Comment (KingDarBoja) [reply src/services/compositionRootService.ts]: Nope, it is not giving any test error if using the `-w` flag.

Comment (JimiC) [reply src/services/compositionRootService.ts]: You need to be more specific. As you can see the tests are passing in all ci suites and on my local machines.

**Comment by JimiC** [comment]
Please hold the merge until I can complete the improvement on PAD.

**Comment by JimiC** [comment]
PAD has been improved. It can now switch between any projects with needing to pass through the default icon manifest state,

Merge approved.

File: src/services/compositionRootService.ts (hunk 1 above)
Comment (KingDarBoja) [reply src/services/compositionRootService.ts]: I was running the test using my Windows 10 OS, five test cases failed if running the `test` command without the `-w` flag. Looks like the related issue wasn't related at all, since the `watch` mode is giving me all test passing.

Comment (JimiC) [reply src/services/compositionRootService.ts]: Just test it on a `Windows 10 1809`. Everything checks out. Also, test it on `MacOS High Sierra` and `Ubuntu 18.04` and all tests pass.

You need to troubleshoot your installation. I suggest to do a clean `npm i` and then `npm t`.

Comment (JimiC) [reply src/services/compositionRootService.ts]: Can anyone else confirm @KingDarBoja issue?

Comment (JimiC) [reply src/services/compositionRootService.ts]: I accidentally stumbled on the issue. The error is presented only when you use `Launch Tests` that uses the `vscode` test runner.

**Comment by JimiC** [comment]
Code review is difficult with all these changes. What I really want is for you to test that the functionality is not broken.

**Comment by JimiC** [comment]
Not possible atm. I'm out of town. If you can resolve the conflicts fine otherwise let it and I will see it on Monday.

**Comment by JimiC** [comment]
Now I saw that the conflicting file is the French translation. It's trivial to resolve it. The new place of the translation files are under `lang` folder. If it's not much to ask someone commit the fix.

**Comment by robertohuertasm** [comment]
I tried to do a `rebase` but it was not as easy as I thought. I prefer you do it as I ended up with strange translation files. I'm not sure to understand the naming change so I'll leave it to you.

**Comment by JimiC** [comment]
OK no problem

**Comment by JimiC** [comment]
Done.
